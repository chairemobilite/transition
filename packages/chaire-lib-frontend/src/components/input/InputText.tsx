/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import React from 'react';

export type InputTextProps = {
    id: string;
    onValueChange?: React.ChangeEventHandler<HTMLTextAreaElement>;
    rows?: number;
    value?: string;
    placeholder?: string;
    maxLength?: number;
    disabled?: boolean;
};

type defaultInputType = {
    name: string;
    id: string;
    maxLength: number;
    value?: string;
    rows?: number;
    onChange?: React.ChangeEventHandler;
    disabled?: boolean;
    placeholder?: string;
};

const InputText: React.FunctionComponent<InputTextProps> = ({
    id,
    onValueChange = undefined,
    rows = 5,
    value = undefined,
    maxLength = 2000,
    disabled = false,
    placeholder
}: InputTextProps) => {
    const defaultAttributes: defaultInputType = {
        rows: rows,
        name: id,
        id: id,
        disabled: disabled === true ? true : false,
        onChange: onValueChange,
        maxLength: maxLength,
        value: value || ''
    };

    if (placeholder) {
        defaultAttributes.placeholder = placeholder;
    }

    return (
        <textarea
            autoComplete="none"
            className="tr__form-input-textarea apptr__input _input-textarea"
            {...defaultAttributes}
        ></textarea>
    );
};

export default InputText;
