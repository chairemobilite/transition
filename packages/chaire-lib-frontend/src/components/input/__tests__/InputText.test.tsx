/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import InputText from '../InputText';

const mockOnChange = jest.fn();
const testId = 'textAreaWidgetId';
const testLabel = 'Text Area';

test('Default props', () => {
    const { container } = render(<InputText
        id = {testId}
    />);
    expect(container).toMatchSnapshot();
});

test('All props', () => {
    const { container } = render(<InputText
        id = {testId}
        onValueChange = {mockOnChange}
        rows = {10}
        value = "test"
        maxLength = {100}
        disabled = {false}
        placeholder = "placeholder"
    />);
    expect(container).toMatchSnapshot();
});

test('Disabled', () => {
    const { container } = render(<InputText
        id = {testId}
        onValueChange = {mockOnChange}
        disabled = {true}
    />);
    expect(container).toMatchSnapshot();
});

test('Placeholder', () => {
    const { container } = render(<InputText
        id = {testId}
        onValueChange = {mockOnChange}
        rows = {10}
        maxLength = {100}
        disabled = {false}
        placeholder = "placeholder"
    />);
    expect(container).toMatchSnapshot();
});

test('Call onChange', () => {
    mockOnChange.mockClear();
    const { getByLabelText } = render(
        <div>
            <InputText
                id = {testId}
                onValueChange = {mockOnChange}
            />
            <label htmlFor={testId}>{testLabel}</label>
        </div>);
    const input = getByLabelText(testLabel) as HTMLInputElement;
    const newText = 'new text';
    fireEvent.change(input, { target: { value: newText } });
    expect(mockOnChange).toHaveBeenCalledTimes(1);
});
