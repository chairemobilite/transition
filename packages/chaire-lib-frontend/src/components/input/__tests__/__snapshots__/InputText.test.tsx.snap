// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All props 1`] = `
<div>
  <textarea
    autocomplete="none"
    class="tr__form-input-textarea apptr__input _input-textarea"
    id="textAreaWidgetId"
    maxlength="100"
    name="textAreaWidgetId"
    placeholder="placeholder"
    rows="10"
  >
    test
  </textarea>
</div>
`;

exports[`Default props 1`] = `
<div>
  <textarea
    autocomplete="none"
    class="tr__form-input-textarea apptr__input _input-textarea"
    id="textAreaWidgetId"
    maxlength="2000"
    name="textAreaWidgetId"
    rows="5"
  />
</div>
`;

exports[`Disabled 1`] = `
<div>
  <textarea
    autocomplete="none"
    class="tr__form-input-textarea apptr__input _input-textarea"
    disabled=""
    id="textAreaWidgetId"
    maxlength="2000"
    name="textAreaWidgetId"
    rows="5"
  />
</div>
`;

exports[`Placeholder 1`] = `
<div>
  <textarea
    autocomplete="none"
    class="tr__form-input-textarea apptr__input _input-textarea"
    id="textAreaWidgetId"
    maxlength="100"
    name="textAreaWidgetId"
    placeholder="placeholder"
    rows="10"
  />
</div>
`;
